projects/igniteui-angular/src/lib/core/utils.ts

```diff
@@ -579,14 +580,113 @@ export const isConstructor = (ref: any) => typeof ref === 'function' && Boolean(
  * Similar to Angular's formatDate. However it will not throw on `undefined | null | ''` instead
  * coalescing to an empty string.
  */
-export const formatDate = (value: string | number | Date, format: string, locale: string, timezone?: string): string => {
+export function formatDate(value: Date | string | number | null | undefined, format: string, locale: string, timezone?: string): string {
     if (value === null || value === undefined || value === '') {
         return '';
     }
-    return _formatDate(value, format, locale, timezone);
-};
+    if (typeof value === "string" || typeof value === "number") {
+        value = getI18nManager().createDateFromValue(value);
+    }
+    let dateStyle = undefined, timeStyle = undefined;
+    if (format === 'short' || format === 'medium' || format === 'long' || format === 'full') {
+        dateStyle = format;
+        timeStyle = format;
+    } else if (format.includes('Date')) {
+        dateStyle = format.replace('Date', '');
+    } else if (format.includes('Time')) {
+        dateStyle = format.replace('Time', '');
+    } else {
+        return getI18nManager().formatDateCustomFormat(value, locale, format, timezone);
+    }
+    const options: Intl.DateTimeFormatOptions = {
+        dateStyle,
+        timeStyle,
+        timeZone: timezone
+    };
+    return getI18nManager().formatDateTime(value, locale, options);
+}
+
+function parseDigitsInfo(value: string) {
+    let minIntegerDigits = undefined, minFractionDigits = undefined, maxFractionDigits = undefined;
+    if (value) {
+        const parts = value.split("-");
+        const innerParts = parts[0].split(".");
+        if (innerParts[0] !== "1") {
+            minIntegerDigits = parseInt(innerParts[0]);
+        }
+        if (innerParts.length == 2 && innerParts[1] !== "0") {
+            minFractionDigits = parseInt(innerParts[1]);
+        }
+        if (parts.length == 2 && parts[1] !== "3") {
+            maxFractionDigits = parseInt(parts[1]);
+        }
+    }
+    return { minIntegerDigits, minFractionDigits, maxFractionDigits };
+}
+
+function formatNumberGeneric(value: number | string | null | undefined, style?: 'decimal' | 'percent' | 'currency', locale?: string, digitsInfo?: string, currencyCode?: string, display?: 'code' | 'symbol' | 'symbol-narrow' | string) {
+    if (value === null || value === undefined || value === '') {
+        return '';
+    }
+    if (typeof value === "string") {
+        value = parseFloat(value);
+    }
+    const parsedDigitsInfo = parseDigitsInfo(digitsInfo);
+    let currencyDisplay: keyof Intl.NumberFormatOptionsCurrencyDisplayRegistry;
+    if (display !== 'code' && display !== 'symbol' && display !== 'symbol-narrow' && display !== 'narrowSymbol' && display !== "name") {
+        currencyDisplay = 'symbol';
+    } else if (display === 'symbol-narrow') {
+        currencyDisplay = 'narrowSymbol';
+    } else {
+        currencyCode = display || undefined;
```

Comment:
Shouldn't this be `currencyDisplay`?  It's setting the code, which is  a different thing.
So if you set a pipeArg, for example:

```
   public pipeArgs: IColumnPipeArgs = {
        display: 'code',
        currencyCode: "EUR"
    };
```

It would generate incorrect options:

<img width="747" height="247" alt="image" src="https://github.com/user-attachments/assets/dac51dd0-b82a-43f7-9e29-4702285b0ff8" />

So it would throw error: RangeError: Invalid currency code : code